icicle/tests/test_field_api.cpp

```diff
@@ -939,6 +942,128 @@ TYPED_TEST(FieldApiTest, ntt)
 }
 #endif // NTT
 
+// define program
+using MlePoly = Symbol<scalar_t>;
+MlePoly combine_func(std::vector<MlePoly>& inputs)
+{
+  const MlePoly& A = inputs[0];
+  const MlePoly& B = inputs[1];
+  const MlePoly& C = inputs[2];
+  const MlePoly& EQ = inputs[3];
+  return (EQ * (A * B - C));
+}
+
+TEST_F(FieldApiTestBase, CpuProgramExecutor)
+{
+  int seed = time(0);
+  srand(seed);
+  ICICLE_LOG_DEBUG << "seed = " << seed;
+
+  // randomize input vectors
+  const int total_size = 100000;
+  auto in_a = std::make_unique<scalar_t[]>(total_size);
+  scalar_t::rand_host_many(in_a.get(), total_size);
+  auto in_b = std::make_unique<scalar_t[]>(total_size);
+  scalar_t::rand_host_many(in_b.get(), total_size);
+  auto in_c = std::make_unique<scalar_t[]>(total_size);
+  scalar_t::rand_host_many(in_c.get(), total_size);
+  auto in_eq = std::make_unique<scalar_t[]>(total_size);
+  scalar_t::rand_host_many(in_eq.get(), total_size);
+
+  //----- element wise operation ----------------------
+  auto out_element_wise = std::make_unique<scalar_t[]>(total_size);
+  std::ostringstream oss;
+  oss << "Strait forward function (Element wise) time: ";
+
+  START_TIMER(element_wise_op)
+  for (int i = 0; i < 100000; ++i) {
+    out_element_wise[i] = in_eq[i] * (in_a[i] * in_b[i] - in_c[i]);
+  }
+  // std::cout << "Strait forward function (Element wise) time: ";
+  END_TIMER(element_wise_op, oss.str().c_str(), true);
+
+  //----- written program ----------------------
+  Program<scalar_t> program_written(&combine_func, 4);
+  // program_written.print_program();
+
+  CpuProgramExecutor<scalar_t> prog_exe_written(program_written);
+  auto out_written_program = std::make_unique<scalar_t[]>(total_size);
+
+  // init program
+  prog_exe_written.m_variable_ptrs[0] = in_a.get();
+  prog_exe_written.m_variable_ptrs[1] = in_b.get();
+  prog_exe_written.m_variable_ptrs[2] = in_c.get();
+  prog_exe_written.m_variable_ptrs[3] = in_eq.get();
+  prog_exe_written.m_variable_ptrs[4] = out_written_program.get();
```

Comment:
is it implicit that last one is output? what if there are multiple outputs?

Reply:
currently only 1 output.
Next step is to support multiple outputs.
Anyway the outputs arrive after the inputs